### ISSUE TITLE
import: IMPORT INTO on an empty table fails on a checkForKeyCollisions error

### ISSUE THREAD

**Issue by msbutler** [cockroachdb/cockroach#97219]
In the [linked](https://github.com/cockroachlabs/support/issues/2051) escalation, we confirmed that an IMPORT INTO job which loaded around 5 billion rows into a  table failed on a `CheckForKeyCollisions` error. Notably, the table was **empty** and had a non visible primary key generated by `gen_random_uuid()`.  This failure implies that we generated identical keys for two _different_ imported rows (different values).  We're not yet sure why, but it likely related to how import overrides the unique ID generation process, described [here](https://github.com/cockroachdb/cockroach/blob/6d22e25358435e4aa9e065c7c8a31fdfdb91c6ee/pkg/sql/row/expr_walker.go#L217).  Note, to generate _uuid_'s import seeds uuid generation with a `unique_rowid`. The codepath is described [here.](https://cockroachlabs.slack.com/archives/C0168LW5THS/p1675966988684039?thread_ts=1675796933.385789&cid=C0168LW5THS) 

We understand the root cause of a related error where a `CheckForKeyCollisions` error fails the second of  _two_ sequential imports  on a table, described [here](https://github.com/cockroachdb/cockroach/issues/96885).

Accompanying slack [thread](https://cockroachlabs.slack.com/archives/C0168LW5THS/p1675796933385789).

**Comment by rafiss** [comment]
@ecwall had a theory about how the overriden `gen_random_uuid` function might have too little entropy, which makes it more likely for collisions to occur if you generate enough rows.

One test we should do here is to run a few 5 billion row imports of our own.

We should also make sure we understand how much entropy is in this function, and see if we can increase the number of random bits.

**Comment by mgartner** [comment]
Is this a duplicate of #96885?

**Comment by msbutler** [comment]
not quite. this issue describes a failure mode of an import into an empty table, whereas https://github.com/cockroachdb/cockroach/issues/96885 describes a failed import into a non empty table that just underwent a successful import.
